In [1]:
import pandas as pd
import numpy as np

In [2]:
#Conjunto de datos de la Tabla Vivienda de 2016
vivienda_2016 = '../../Datasets/ENH/dataframes/vivienda_2016.csv'

df_vivienda_2016 = pd.read_csv(vivienda_2016, low_memory=False, index_col=0)

df_vivienda_2016.reset_index(inplace=True)

df_vivienda_2016.head()

index   folioviv tipo_viv condominio elevador  mat_pared  mat_techos  \
0      0  100006018        1                              8           9   
1      1  100006037        1                              8           9   
2      2  100006055        1                              8           9   
3      3  100006068        1                              8           9   
4      4  100006085        1                              8           9   

   mat_pisos  ais_techos  ais_pared   ...     ubica_geo   ageb tam_loc  \
0          3           2          2   ...      10010001  028-6       1   
1          3           2          2   ...      10010001  028-6       1   
2          3           2          2   ...      10010001  028-6       1   
3          3           2          2   ...      10010001  028-6       1   
4          2           2          2   ...      10010001  028-6       1   

   est_socio est_dis  upm  factor  CVE_LOC CVE_MUN  CVE_ENT  
0          3       1    1     165        1      35       32  
1          3       1    1     165        1      35       32  
2          3       1    1     165        1      35       32  
3          3       1    1     165        1      35       32  
4          3       1    1     165        1      35       32  

[5 rows x 113 columns]

In [3]:
#Conjunto de datos de la Tabla Vivienda de 2015
vivienda_2015 = '../../Datasets/ENH/dataframes/vivienda_2015.csv'

df_vivienda_2015 = pd.read_csv(vivienda_2015, low_memory=False, index_col=0)

df_vivienda_2015.reset_index(inplace=True)

df_vivienda_2015.head()

index  folioviv tipo_viv  mat_pared  mat_techos mat_pisos antiguedad  \
0      0  10000601        1          8           9         3              
1      1  10000602        1          8           9         3         25   
2      2  10000603        1          8           9         3              
3      3  10000604        1          8           9         3              
4      4  10000605        1          8           9         3         40   

   cocina cocina_dor  cuart_dorm   ...     ubica_geo   ageb tam_loc  \
0       1          2           2   ...      10010001  028-6       1   
1       1          2           1   ...      10010001  028-6       1   
2       1          2           4   ...      10010001  028-6       1   
3       1          2           2   ...      10010001  028-6       1   
4       1          2           2   ...      10010001  028-6       1   

   est_socio est_dis upm factor_viv CVE_LOC CVE_MUN  CVE_ENT  
0          3       5  10        163       1      37       32  
1          3       5  10        163       1      37       32  
2          3       5  10        163       1      37       32  
3          3       5  10        163       1      37       32  
4          3       5  10        163       1      37       32  

[5 rows x 89 columns]

In [4]:
#Conjunto de datos de la Tabla Vivienda de 2014
vivienda_2014 = '../../Datasets/ENH/dataframes/vivienda_2014.csv'

df_vivienda_2014 = pd.read_csv(vivienda_2014, low_memory=False, index_col=0)

df_vivienda_2014.reset_index(inplace=True)

df_vivienda_2014.head()

index  folioviv tipo_viv  mat_pared  mat_techos  mat_pisos  \
0      0  10001801        1          8           9          3   
1      1  10001803        1          8           9          3   
2      2  10001804        1          8           9          3   
3      3  10001805        1          8           9          3   
4      4  10002201        1          8           9          3   

              antiguedad  cocina  cocina_dor  cuart_dorm   ...     ubica_geo  \
0                     30       1         2.0           3   ...      10010001   
1                     26       1         2.0           3   ...      10010001   
2                     23       1         2.0           2   ...      10010001   
3                     10       1         2.0           4   ...      10010001   
4  1.79769313486232e+308       1         2.0           1   ...      10010001   

    ageb  tam_loc  est_socio  est_dis   upm factor_viv CVE_LOC CVE_MUN  \
0  029-0        1          3        9   710        243       1      37   
1  029-0        1          3        9   710        243       1      37   
2  029-0        1          3        9   710        243       1      37   
3  029-0        1          3        9   710        243       1      37   
4  030-3        1          3        2  1450        262       1      37   

   CVE_ENT  
0       32  
1       32  
2       32  
3       32  
4       32  

[5 rows x 69 columns]

In [5]:
frames_to_concat = [df_vivienda_2014, df_vivienda_2015, df_vivienda_2016]

In [6]:
df_vivienda = pd.concat(frames_to_concat, ignore_index=True)
df_vivienda.head()

CVE_ENT  CVE_LOC  CVE_MUN   ageb  aire_acond  ais_otro  ais_pared  \
0       32        1       37  029-0           2       NaN        NaN   
1       32        1       37  029-0           2       NaN        NaN   
2       32        1       37  029-0           2       NaN        NaN   
3       32        1       37  029-0           2       NaN        NaN   
4       32        1       37  030-3           2       NaN        NaN   

   ais_techos  ais_ventan anio_panel    ...     tipo_viv  tot_hog tot_resid  \
0         NaN         NaN        NaN    ...            1        1         4   
1         NaN         NaN        NaN    ...            1        1         4   
2         NaN         NaN        NaN    ...            1        1         2   
3         NaN         NaN        NaN    ...            1        1         6   
4         NaN         NaN        NaN    ...            1        1         4   

  tv_paga ubica_geo   upm  uso_compar  valor_viv  vend_hered  viv_usada  
0     NaN  10010001   710           2        NaN         NaN        NaN  
1     NaN  10010001   710           2        NaN         NaN        NaN  
2     NaN  10010001   710           2        NaN         NaN          2  
3     NaN  10010001   710           2        NaN         NaN        NaN  
4     NaN  10010001  1450           2        NaN         NaN          1  

[5 rows x 116 columns]

In [121]:
df_copy_vivienda = df_vivienda.copy()
df_copy_vivienda.head()

CVE_ENT  CVE_LOC  CVE_MUN   ageb  aire_acond  ais_otro  ais_pared  \
0       32        1       37  029-0           2       NaN        NaN   
1       32        1       37  029-0           2       NaN        NaN   
2       32        1       37  029-0           2       NaN        NaN   
3       32        1       37  029-0           2       NaN        NaN   
4       32        1       37  030-3           2       NaN        NaN   

   ais_techos  ais_ventan anio_panel    ...     tipo_viv  tot_hog tot_resid  \
0         NaN         NaN        NaN    ...            1        1         4   
1         NaN         NaN        NaN    ...            1        1         4   
2         NaN         NaN        NaN    ...            1        1         2   
3         NaN         NaN        NaN    ...            1        1         6   
4         NaN         NaN        NaN    ...            1        1         4   

  tv_paga ubica_geo   upm  uso_compar  valor_viv  vend_hered  viv_usada  
0     NaN  10010001   710           2        NaN         NaN        NaN  
1     NaN  10010001   710           2        NaN         NaN        NaN  
2     NaN  10010001   710           2        NaN         NaN          2  
3     NaN  10010001   710           2        NaN         NaN        NaN  
4     NaN  10010001  1450           2        NaN         NaN          1  

[5 rows x 116 columns]

In [166]:
for col in df_copy_vivienda.columns:
    try:
        df_copy_vivienda[col].fillna(value='&', inplace=True)
    except:
        pass
    
df_copy_vivienda.head()

CVE_ENT  CVE_LOC  CVE_MUN   ageb  aire_acond ais_otro ais_pared ais_techos  \
0      32        1       37  029-0           2        &         &          &   
1      32        1       37  029-0           2        &         &          &   
2      32        1       37  029-0           2        &         &          &   
3      32        1       37  029-0           2        &         &          &   
4      32        1       37  030-3           2        &         &          &   

  ais_ventan anio_panel    ...    tipo_viv tot_hog tot_resid tv_paga  \
0          &          &    ...           1       1         4       &   
1          &          &    ...           1       1         4       &   
2          &          &    ...           1       1         2       &   
3          &          &    ...           1       1         6       &   
4          &          &    ...           1       1         4       &   

  ubica_geo   upm  uso_compar valor_viv vend_hered viv_usada  
0  10010001   710           2         &          &         &  
1  10010001   710           2         &          &         &  
2  10010001   710           2         &          &         2  
3  10010001   710           2         &          &         &  
4  10010001  1450           2         &          &         1  

[5 rows x 116 columns]

In [178]:
for col in df_copy_vivienda.columns:
    try:
        df_copy_vivienda[col].replace({' ' : '&'}, inplace=True)
    except:
        pass
    
df_copy_vivienda.head()

CVE_ENT  CVE_LOC  CVE_MUN   ageb aire_acond ais_otro ais_pared ais_techos  \
0      32        1       37  029-0          2        &         &          &   
1      32        1       37  029-0          2        &         &          &   
2      32        1       37  029-0          2        &         &          &   
3      32        1       37  029-0          2        &         &          &   
4      32        1       37  030-3          2        &         &          &   

  ais_ventan anio_panel    ...    tipo_viv tot_hog tot_resid tv_paga  \
0          &          &    ...           1       1         4       &   
1          &          &    ...           1       1         4       &   
2          &          &    ...           1       1         2       &   
3          &          &    ...           1       1         6       &   
4          &          &    ...           1       1         4       &   

  ubica_geo   upm uso_compar valor_viv vend_hered viv_usada  
0  10010001   710          2         &          &         &  
1  10010001   710          2         &          &         &  
2  10010001   710          2         &          &         2  
3  10010001   710          2         &          &         &  
4  10010001  1450          2         &          &         1  

[5 rows x 116 columns]

In [180]:
for col in df_copy_vivienda.columns:
    try:
        df_copy_vivienda[col].replace({'1.79769313486232e+308' : '&'}, inplace=True)
    except:
        pass
    
df_copy_vivienda.head()

CVE_ENT  CVE_LOC  CVE_MUN   ageb aire_acond ais_otro ais_pared ais_techos  \
0      32        1       37  029-0          2        &         &          &   
1      32        1       37  029-0          2        &         &          &   
2      32        1       37  029-0          2        &         &          &   
3      32        1       37  029-0          2        &         &          &   
4      32        1       37  030-3          2        &         &          &   

  ais_ventan anio_panel    ...    tipo_viv tot_hog tot_resid tv_paga  \
0          &          &    ...           1       1         4       &   
1          &          &    ...           1       1         4       &   
2          &          &    ...           1       1         2       &   
3          &          &    ...           1       1         6       &   
4          &          &    ...           1       1         4       &   

  ubica_geo   upm uso_compar valor_viv vend_hered viv_usada  
0  10010001   710          2         &          &         &  
1  10010001   710          2         &          &         &  
2  10010001   710          2         &          &         2  
3  10010001   710          2         &          &         &  
4  10010001  1450          2         &          &         1  

[5 rows x 116 columns]

## Análisis

#### Tipo de Vivienda `tipo_viv`
Definición: Clase de vivienda particular

Valor Etiqueta:
+ 1 Casa independiente
+ 2 Departamento en edificio
+ 3 Vivienda en vecindad
+ 4 Vivienda en cuarto de azotea
+ 5 Local no construido para habitación
+ & representa un dato “No especificado”


In [50]:
df_copy_vivienda.tipo_viv = df_copy_vivienda.tipo_viv.astype('category')
df_copy_vivienda.tipo_viv.value_counts()

1    153405
2      5310
3      1524
5       104
&        71
4        53
Name: tipo_viv, dtype: int64

#### Número de pisos en el condominio `condominio`
Definición: Número de pisos que tiene el departamento en condominio vertical.

Valor Etiqueta:
+ 1+


In [186]:
df_copy_vivienda.condominio = df_copy_vivienda.condominio.astype('category')
df_copy_vivienda.condominio.value_counts().head()

&    158516
2       738
4       503
3       494
5       128
Name: condominio, dtype: int64

#### Disponibilidad de elevador `elevador`
Definición: Posee elevador el departamento en condominio vertical.

Valor Etiqueta:
+ 1 Sí
+ 2 No


In [187]:
df_copy_vivienda.elevador = df_copy_vivienda.elevador.astype('category')
df_copy_vivienda.elevador.value_counts().head()

&    158516
2      1812
1       139
Name: elevador, dtype: int64

#### Material de paredes `mat_pared`
Definición: Material predominante en las paredes de la vivienda. 
    
Valor Etiqueta
+ 1 Material de desecho
+ 2 Lámina de cartón
+ 3 Lámina de asbesto o metálica
+ 4 Carrizo, bambú o palma
+ 5 Embarro o bajareque
+ 6 Madera
+ 7 Adobe
+ 8 Tabique, ladrillo, block, piedra, cantera, cemento o concreto

In [51]:
df_copy_vivienda.mat_pared = df_copy_vivienda.mat_pared.astype('category')
df_copy_vivienda.mat_pared.value_counts()

8    140992
7     10988
6      5943
3       913
5       649
4       550
2       235
1       197
Name: mat_pared, dtype: int64

#### Material de techos `mat_techos`
Definición: Material predominante en el techo de la vivienda.
    
Valor Etiqueta
+ 1 Material de desecho
+ 2 Lámina de cartón
+ 3 Lámina metálica
+ 4 Lámina de asbesto
+ 5 Palma o paja
+ 6 Madera o tejamanil
+ 7 Terrado con viguería
+ 8 Teja
+ 9 Losa de concreto o viguetas con bovedilla

In [184]:
df_copy_vivienda.mat_techos = df_copy_vivienda.mat_techos.astype('category')
df_copy_vivienda.mat_techos.value_counts()

9    118950
3     21935
4      8846
6      3649
8      2573
7      1753
2      1709
5       938
1       114
Name: mat_techos, dtype: int64

#### Material de pisos `mat_pisos`
Definición: Material predominante en el piso de la vivienda.
    
Valor Etiqueta
+ 1 Tierra
+ 2 Cemento o firme
+ 3 Madera, mosaico u otro recubrimiento
+ & representa un dato “No especificado”

In [52]:
df_copy_vivienda.mat_pisos = df_copy_vivienda.mat_pisos.astype('category')
#df_copy_vivienda.groupby('mat_pisos').mat_pisos.count()
df_copy_vivienda.mat_pisos.value_counts()

2    81341
3    74942
1     4183
&        1
Name: mat_pisos, dtype: int64

In [53]:
# hay 7 categorías donde debería haber 4, entonces se crea un diccionario para solucionarlo
cat_mat_pisos = {'1': 1, '2': 2, '3': 3, '&': '&',
                 1: 1, 2: 2, 3: 3}

#type(cat_mat_pisos)
cat_mat_pisos

{'&': '&', '1': 1, 1: 1, '2': 2, 2: 2, '3': 3, 3: 3}

In [54]:
df_copy_vivienda.mat_pisos.replace(cat_mat_pisos, inplace=True)
df_copy_vivienda.mat_pisos.value_counts()

2    81341
3    74942
1     4183
&        1
Name: mat_pisos, dtype: int64

#### Aislamiento en techo `ais_techos`
Definición: Aislamiento en el techo de la vivienda.
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & No sabe

In [188]:
df_copy_vivienda.ais_techos = df_copy_vivienda.ais_techos.astype('category')
df_copy_vivienda.ais_techos.value_counts()

&      102401
2.0     54987
1.0      2624
9.0       455
Name: ais_techos, dtype: int64

In [190]:
cat_ais_techos = {'1.0': 1, 1.0 : 1,
                  '2.0': 2, 2.0 : 2,
                  '9.0' : '&', 9.0 : '&', 9 : '&'}

#type(cat_mat_pisos)
cat_ais_techos

{'1.0': 1, 1.0: 1, '2.0': 2, 2.0: 2, '9.0': '&', 9.0: '&'}

In [191]:
df_copy_vivienda.ais_techos.replace(cat_ais_techos, inplace=True)
df_copy_vivienda.ais_techos.value_counts()

&    102856
2     54987
1      2624
Name: ais_techos, dtype: int64

#### Aislamiento en paredes `ais_pared`
Definición: Aislamiento en las paredes de la vivienda.
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & No sabe

In [192]:
df_copy_vivienda.ais_pared = df_copy_vivienda.ais_pared.astype('category')
df_copy_vivienda.ais_pared.value_counts()

&      102401
2.0     57040
1.0       620
9.0       406
Name: ais_pared, dtype: int64

In [193]:
cat_ais_pared = {'1.0': 1, 1.0 : 1,
                  '2.0': 2, 2.0 : 2,
                  '9.0' : '&', 9.0 : '&', 9 : '&'}

cat_ais_pared

{'1.0': 1, 1.0: 1, '2.0': 2, 2.0: 2, '9.0': '&', 9.0: '&'}

In [194]:
df_copy_vivienda.ais_pared.replace(cat_ais_pared, inplace=True)
df_copy_vivienda.ais_pared.value_counts()

&    102807
2     57040
1       620
Name: ais_pared, dtype: int64

#### Aislamiento en ventanas `ais_ventan`
Definición: Aislamiento en las ventanas de la vivienda.
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & No sabe

In [195]:
df_copy_vivienda.ais_ventan = df_copy_vivienda.ais_ventan.astype('category')
df_copy_vivienda.ais_ventan.value_counts()

&      102401
2.0     57321
1.0       404
9.0       341
Name: ais_ventan, dtype: int64

In [196]:
cat_ais_ventan = {'1.0': 1, 1.0 : 1,
                  '2.0': 2, 2.0 : 2,
                  '9.0' : '&', 9.0 : '&', 9 : '&'}

cat_ais_ventan

{'1.0': 1, 1.0: 1, '2.0': 2, 2.0: 2, '9.0': '&', 9.0: '&'}

In [198]:
df_copy_vivienda.ais_ventan.replace(cat_ais_ventan, inplace=True)
df_copy_vivienda.ais_ventan.value_counts()

&    102742
2     57321
1       404
Name: ais_ventan, dtype: int64

#### Otro tipo de aislamiento `ais_otro`
Definición: Otro tipo de aislamiento en la vivienda. 
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & No sabe

In [200]:
df_copy_vivienda.ais_otro = df_copy_vivienda.ais_otro.astype('category')
df_copy_vivienda.ais_otro.value_counts()

&      102401
2.0     57633
9.0       332
1.0       101
Name: ais_otro, dtype: int64

In [201]:
cat_ais_otro = {'1.0': 1, 1.0 : 1,
                  '2.0': 2, 2.0 : 2,
                  '9.0' : '&', 9.0 : '&', 9 : '&'}

cat_ais_otro

{'1.0': 1, 1.0: 1, '2.0': 2, 2.0: 2, '9.0': '&', 9.0: '&'}

In [202]:
df_copy_vivienda.ais_otro.replace(cat_ais_otro, inplace=True)
df_copy_vivienda.ais_otro.value_counts()

&    102733
2     57633
1       101
Name: ais_otro, dtype: int64

#### Antigüedad de la vivienda `antiguedad`
Definición: Tiempo que tiene la vivienda de haber sido construida.
    
Valor Etiqueta
+ 0 - 99 Años
+ 0 Menos de un año 
+ & No sabe

In [55]:
df_copy_vivienda.antiguedad = df_copy_vivienda.antiguedad.astype('category')
#df_copy_vivienda.groupby('mat_pisos').mat_pisos.count()
df_copy_vivienda.antiguedad.value_counts(sort=True, ascending=False).head()

&     40601
20    10142
10     9988
30     8666
15     8240
Name: antiguedad, dtype: int64

In [56]:
# hay 7 categorías donde debería haber 4, entonces se crea un diccionario para solucionarlo
cat_antiguedad = {'1.79769313486232e+308': '&'}

#type(cat_mat_pisos)
cat_antiguedad

{'1.79769313486232e+308': '&'}

In [69]:
df_copy_vivienda.antiguedad.replace(cat_antiguedad, inplace=True)
df_copy_vivienda.antiguedad.value_counts(sort=True, ascending=False).head()

&     40601
20    10142
10     9988
30     8666
15     8240
Name: antiguedad, dtype: int64

#### Tiene cocina `cocina` 
Definición: Cuarto de la vivienda dispuesto para preparar o calentar alimentos, aunque también tenga otros usos.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [18]:
df_copy_vivienda.cocina = df_copy_vivienda.cocina.astype('category')
#df_copy_vivienda.groupby('mat_pisos').mat_pisos.count()
df_copy_vivienda.cocina.value_counts()

1    151424
2      9043
Name: cocina, dtype: int64

#### Utiliza cocina de dormitorio `cocina_dor` 
Definición: Cocina utilizada de manera habitual también para dormir.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [64]:
df_copy_vivienda.cocina_dor = df_copy_vivienda.cocina_dor.astype('category')
#df_copy_vivienda.groupby('mat_pisos').mat_pisos.count()
df_copy_vivienda.cocina_dor.value_counts()

2    139982
1     11442
&      6455
Name: cocina_dor, dtype: int64

In [62]:
cat_cocina_dor = {'1': 1, '1.0' : 1, 1 : 1,
                  '2': 2, '2.0' : 2, 2 : 2,
                  ' ' : '&'}
cat_cocina_dor

{' ': '&', '1': 1, 1: 1, '1.0': 1, '2': 2, 2: 2, '2.0': 2}

In [63]:
df_copy_vivienda.cocina_dor.replace(cat_cocina_dor, inplace=True)
df_copy_vivienda.cocina_dor.value_counts()

2    139982
1     11442
&      6455
Name: cocina_dor, dtype: int64

#### Cuartos dormitorio `cuart_dorm`
Definición: Número de cuartos de la vivienda que son usados habitualmente para dormir, aunque también tengan otros usos.
    
Valor Etiqueta
+ 1+

In [20]:
df_copy_vivienda.cuart_dorm = df_copy_vivienda.cuart_dorm.astype('category')
#df_copy_vivienda.groupby('mat_pisos').mat_pisos.count()
df_copy_vivienda.cuart_dorm.value_counts()

2     67399
1     53148
3     31868
4      6669
5      1071
6       228
7        55
8        25
9         2
11        1
10        1
Name: cuart_dorm, dtype: int64

#### Número de cuartos `num_cuarto` 
Definición: Número total de cuartos que tiene la vivienda, independientemente de su uso.
    
Valor Etiqueta
+ 1+

In [21]:
df_copy_vivienda.num_cuarto = df_copy_vivienda.num_cuarto.astype('category')
#df_copy_vivienda.groupby('mat_pisos').mat_pisos.count()
df_copy_vivienda.num_cuarto.value_counts()

4     42993
3     37482
5     28022
2     21526
6     12397
1      9187
7      4926
8      2377
9       846
10      432
11      124
12       95
13       29
14       11
15        9
16        5
17        2
22        2
20        1
18        1
Name: num_cuarto, dtype: int64

#### Disponibilidad de agua `disp_agua` 
Definición: Formas de abastecer el agua a la vivienda ya sea que se cuente o no, con agua entubada de la red pública.
    
Valor Etiqueta
+ 1 Agua entubada dentro de la vivienda
+ 2 Agua entubada fuera de la vivienda, pero dentro del terreno
+ 3 Agua entubada de llave pública (o hidrante)
+ 4 Agua entubada que acarrean de otra vivienda
+ 5 Agua de pipa
+ 6 Agua de un pozo, río, lago, arroyo u otra

In [22]:
df_copy_vivienda.disp_agua = df_copy_vivienda.disp_agua.astype('category')
#df_copy_vivienda.groupby('mat_pisos').mat_pisos.count()
df_copy_vivienda.disp_agua.value_counts()

1    116062
2     33701
6      6081
5      2300
4      1889
3       434
Name: disp_agua, dtype: int64

#### Dotación de agua `dotac_agua` 
Definición: Frecuencia con la que llega a la vivienda el agua entubada que proviene de la red pública.
    
Valor Etiqueta
+ 1 Diario
+ 2 Cada tercer día
+ 3 Dos veces por semana
+ 4 Una vez por semana
+ 5 De vez en cuando

In [23]:
df_copy_vivienda.dotac_agua = df_copy_vivienda.dotac_agua.astype('category')
#df_copy_vivienda.groupby('mat_pisos').mat_pisos.count()
df_copy_vivienda.dotac_agua.value_counts()

1      81030
1.0    29856
2      16169
        7628
2.0     6545
3       5091
4       3864
5       2899
3.0     2053
4.0     1374
5.0      882
Name: dotac_agua, dtype: int64

In [24]:
# hay 12 categorías donde debería haber 5, entonces se crea un diccionario para solucionarlo
cat_dotac_agua = {'1': 1, '1.0': 1, 1 : 1,
                 '2': 2, '2.0': 2, 2 : 2,
                 '3': 3, '3.0': 3, 3 : 3,
                 '4': 3, '4.0': 3, 4 : 3,
                 '5': 3, '5.0': 3, 5 : 3,
                 ' ': '&'}

#type(cat_mat_pisos)
cat_dotac_agua

{' ': '&',
 '1': 1,
 1: 1,
 '1.0': 1,
 '2': 2,
 2: 2,
 '2.0': 2,
 '3': 3,
 3: 3,
 '3.0': 3,
 '4': 3,
 4: 3,
 '4.0': 3,
 '5': 3,
 5: 3,
 '5.0': 3}

In [25]:
df_copy_vivienda.dotac_agua.replace(cat_dotac_agua, inplace=True)
df_copy_vivienda.dotac_agua.value_counts()

1    110886
2     22714
3     16163
&      7628
Name: dotac_agua, dtype: int64

#### Tiene excusado `excusado` 
Definición: Instalación sanitaria o excusado en la vivienda para el desalojo de los desechos humanos.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [26]:
df_copy_vivienda.excusado = df_copy_vivienda.excusado.astype('category')
df_copy_vivienda.excusado.value_counts()

1    154799
2      5668
Name: excusado, dtype: int64

#### Uso compartido del sanitario  `uso_compar`
Definición: Uso compartido de la instalación sanitaria con los habitantes de otra vivienda
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [27]:
df_copy_vivienda.uso_compar = df_copy_vivienda.uso_compar.astype('category')
df_copy_vivienda.uso_compar.value_counts()

2      107407
2.0     40282
1        5057
         4217
1.0      2053
Name: uso_compar, dtype: int64

In [28]:
cat_uso_compar = {'1': 1, '1.0': 1, 1 : 1,
                 '2': 2, '2.0': 2, 2 : 2,
                 ' ': '&'}

#type(cat_mat_pisos)
cat_uso_compar

{' ': '&', '1': 1, 1: 1, '1.0': 1, '2': 2, 2: 2, '2.0': 2}

In [29]:
df_copy_vivienda.uso_compar.replace(cat_uso_compar, inplace=True)
df_copy_vivienda.uso_compar.value_counts()

2    147689
1      7110
&      4217
Name: uso_compar, dtype: int64

#### Sanitario conexión agua `sanit_agua` 
Definición: Funcionamiento de la instalación sanitaria con o sin conexión de agua.
    
Valor Etiqueta
+ 1 Tiene descarga directa de agua
+ 2 Le echan agua con cubeta
+ 3 No se le puede echar agua

In [30]:
df_copy_vivienda.sanit_agua = df_copy_vivienda.sanit_agua.astype('category')
df_copy_vivienda.sanit_agua.value_counts()

1      80198
1.0    29882
2      26948
2.0    10321
3       5318
        4217
3.0     2132
Name: sanit_agua, dtype: int64

In [31]:
cat_sanit_agua = {'1': 1, '1.0': 1, 1 : 1,
                  '2': 2, '2.0': 2, 2 : 2,
                  '3': 3, '3.0': 3, 3 : 3,
                  ' ': '&'}

#type(cat_mat_pisos)
cat_sanit_agua

{' ': '&',
 '1': 1,
 1: 1,
 '1.0': 1,
 '2': 2,
 2: 2,
 '2.0': 2,
 '3': 3,
 3: 3,
 '3.0': 3}

In [32]:
df_copy_vivienda.sanit_agua.replace(cat_sanit_agua, inplace=True)
df_copy_vivienda.sanit_agua.value_counts()

1    110080
2     37269
3      7450
&      4217
Name: sanit_agua, dtype: int64

#### Sanitario excusado regadera `bano_comp`
Definición: Número de baños con excusado y regadera.
    
Valor Etiqueta
+ 1+

In [33]:
df_copy_vivienda.bano_comp = df_copy_vivienda.bano_comp.astype('category')
df_copy_vivienda.bano_comp.value_counts()

1                        86412
0                        42643
2                        20664
                          4217
3                         3886
1.79769313486232e+308     1451
4                          901
5                          216
6                           56
7                           13
8                            5
9                            3
Name: bano_comp, dtype: int64

In [34]:
# hay 12 categorías donde debería haber 5, entonces se crea un diccionario para solucionarlo
cat_bano_comp = {' ': '&', 
                 '1.79769313486232e+308' : '&'}
cat_bano_comp

{' ': '&', '1.79769313486232e+308': '&'}

In [35]:
df_copy_vivienda.bano_comp.replace(cat_bano_comp, inplace=True)
df_copy_vivienda.bano_comp.value_counts()

1    86412
0    42643
2    20664
&     5668
3     3886
4      901
5      216
6       56
7       13
8        5
9        3
Name: bano_comp, dtype: int64

#### Sanitario excusado `bano_excus`
Definición: Número de baños sólo con excusado.
    
Valor Etiqueta
+ 1+

In [36]:
df_copy_vivienda.bano_excus = df_copy_vivienda.bano_excus.astype('category')
df_copy_vivienda.bano_excus.value_counts()

0                        94308
1                        57388
                          4217
2                         2772
1.79769313486232e+308     1451
3                          184
9                           73
4                           41
8                           20
5                            7
6                            6
Name: bano_excus, dtype: int64

In [37]:
cat_bano_excus = {' ': '&', 
                 '1.79769313486232e+308' : '&'}
cat_bano_excus

{' ': '&', '1.79769313486232e+308': '&'}

In [38]:
df_copy_vivienda.bano_excus.replace(cat_bano_excus, inplace=True)
df_copy_vivienda.bano_excus.value_counts()

0    94308
1    57388
&     5668
2     2772
3      184
9       73
4       41
8       20
5        7
6        6
Name: bano_excus, dtype: int64

#### Sanitario regadera `bano_regad`
Definición: Número de baños sólo con regadera.
    
Valor Etiqueta
+ 1+

In [39]:
df_copy_vivienda.bano_regad = df_copy_vivienda.bano_regad.astype('category')
df_copy_vivienda.bano_regad.value_counts()

0                        150563
                           4217
1                          4043
1.79769313486232e+308      1451
9                            83
2                            80
8                            20
3                             9
4                             1
Name: bano_regad, dtype: int64

In [40]:
# hay 12 categorías donde debería haber 5, entonces se crea un diccionario para solucionarlo
cat_bano_regad = {' ': '&', 
                 '1.79769313486232e+308' : '&'}
cat_bano_regad

{' ': '&', '1.79769313486232e+308': '&'}

In [41]:
df_copy_vivienda.bano_regad.replace(cat_bano_regad, inplace=True)
df_copy_vivienda.bano_regad.value_counts()

0    150563
&      5668
1      4043
9        83
2        80
8        20
3         9
4         1
Name: bano_regad, dtype: int64

#### Destino de drenaje `drenaje`
Definición: Disponibilidad de un sistema de drenaje para desalojar de la vivienda los desechos humanos y las aguas utilizadas.
    
Valor Etiqueta
+ 1 La red pública
+ 2 Una fosa séptica
+ 3 Una tubería que va a dar a una barranca o grieta
+ 4 Una tubería que va a dar a un río, lago o mar
+ 5 No tiene drenaje

In [42]:
df_copy_vivienda.drenaje = df_copy_vivienda.drenaje.astype('category')
df_copy_vivienda.drenaje.value_counts()

1    114883
2     33666
5     10283
3      1104
4       531
Name: drenaje, dtype: int64

#### Disponibilidad eléctrica `disp_elect`
Definición: Fuente de donde se obtiene la energía eléctrica en la vivienda.
    
Valor Etiqueta
+ 1 Del servicio público
+ 2 De una planta particular
+ 3 De panel solar
+ 4 De otra fuente
+ 5 No tiene luz eléctrica

In [43]:
df_copy_vivienda.disp_elect = df_copy_vivienda.disp_elect.astype('category')
df_copy_vivienda.disp_elect.value_counts()

1    158964
5      1036
3       317
2       105
4        45
Name: disp_elect, dtype: int64

#### Año panel solar `anio_panel`
Definición: Año en el que adquirieron el panel solar
    
Valor Etiqueta
+ 1+

In [203]:
df_copy_vivienda.anio_panel = df_copy_vivienda.anio_panel.astype('category')
df_copy_vivienda.anio_panel.value_counts(sort=True, ascending=False).head()

&       160379
2015        16
2010        14
2013         8
2014         7
Name: anio_panel, dtype: int64

#### Desconoce año de adquisición `panel_ne`
Definición: Desconoce el año en el año en el que adquirieron el panel solar.
    
Valor Etiqueta
+ & 

In [205]:
df_copy_vivienda.panel_ne = df_copy_vivienda.panel_ne.astype('category')
df_copy_vivienda.panel_ne.value_counts(sort=True, ascending=False).head()

&    160443
9        24
Name: panel_ne, dtype: int64

#### Conocimiento en potencia instalada  `pot_panel`
Definición: Conocimiento referente a la potencia instalada del panel solar. 
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [206]:
df_copy_vivienda.pot_panel = df_copy_vivienda.pot_panel.astype('category')
df_copy_vivienda.pot_panel.value_counts(sort=True, ascending=False).head()

&    160355
2        87
1        25
Name: pot_panel, dtype: int64

#### Potencia instalada `potencia`
Definición: Potencia instalada en el panel solar, medida en kW.
    
Valor Etiqueta
+ 1 Del servicio público
+ 2 De una planta particular
+ 3 De panel solar
+ 4 De otra fuente
+ 5 No tiene luz eléctrica

In [80]:
df_copy_vivienda.potencia = df_copy_vivienda.potencia.astype('category')
df_copy_vivienda.potencia.value_counts(sort=True, ascending=False).head()

&      160301
135        25
120        14
600        13
75         12
Name: potencia, dtype: int64

In [45]:
# hay 12 categorías donde debería haber 5, entonces se crea un diccionario para solucionarlo
cat_potencia = {' ': '&', 
                 '1.79769313486232e+308' : '&',
               '-1' : '&'}
cat_potencia

{' ': '&', '-1': '&', '1.79769313486232e+308': '&'}

In [71]:
df_copy_vivienda.potencia.replace(cat_potencia, inplace=True)
df_copy_vivienda.potencia.value_counts(sort=True, ascending=False).head()

&      160301
135        25
120        14
600        13
75         12
Name: potencia, dtype: int64

#### Número de focos incandescente `focos_inca`
Definición: Número de focos incandescentes que se usan para iluminar la vivienda, tanto en el interior como en el exterior.
    
Valor Etiqueta
+ 1+

In [182]:
df_copy_vivienda.focos_inca = df_copy_vivienda.focos_inca.astype('category')
df_copy_vivienda.focos_inca.value_counts(sort=True, ascending=False).head()

0    82148
2    16433
1    14922
3    13641
4    10977
Name: focos_inca, dtype: int64

#### Número de focos ahorradores `focos_ahor`
Definición: Número de focos ahorradores que se usan para iluminar la vivienda, tanto en el interior como en el exterior.
    
Valor Etiqueta
+ 1+

In [183]:
df_copy_vivienda.focos_ahor = df_copy_vivienda.focos_ahor.astype('category')
df_copy_vivienda.focos_ahor.value_counts(sort=True, ascending=False).head()

0    27730
4    16448
5    15317
3    15185
6    14513
Name: focos_ahor, dtype: int64

#### Tipo de combustible `combustible`
Definición: Combustible más utilizado para preparar o calentar alimentos en la vivienda.
    
Valor Etiqueta
+ 1 Leña
+ 2 Carbón
+ 3 Gas de tanque
+ 4 Gas natural o de tubería
+ 5 Electricidad
+ 6 Otro combustible
+ & representa un dato “No especificado”

In [77]:
df_copy_vivienda.combustible = df_copy_vivienda.combustible.astype('category')
df_copy_vivienda.combustible.value_counts(sort=True, ascending=False)

3    123283
1     25130
4      5110
4      3002
6      1132
5       960
5       704
6       616
2       356
2       172
&         2
Name: combustible, dtype: int64

In [84]:
cat_combustible = {'1' : 1, 1: 1,
                   '2': 2, 2 : 2,
                   '3': 3, 3 : 3,
                   '4' : 4, 4: 4,
                   '5': 5, 5 : 5, 
                   '6': 6, 6 : 6}
cat_combustible

{'1': 1,
 1: 1,
 '2': 2,
 2: 2,
 '3': 3,
 3: 3,
 '4': 4,
 4: 4,
 '5': 5,
 5: 5,
 '6': 6,
 6: 6}

In [86]:
df_copy_vivienda.combustible.replace(cat_combustible, inplace=True)
df_copy_vivienda.combustible.value_counts(sort=True, ascending=False)

3    123283
1     25130
4      8112
6      1748
5      1664
2       528
&         2
Name: combustible, dtype: int64

#### Estufa con chimenea `estufa_chi`
Definición: Presencia de chimenea para desalojar el humo al cocinar.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [87]:
df_copy_vivienda.estufa_chi = df_copy_vivienda.estufa_chi.astype('category')
df_copy_vivienda.estufa_chi.value_counts(sort=True, ascending=False)

       98259
2      14419
2.0     5667
1       4003
1.0     1569
Name: estufa_chi, dtype: int64

In [91]:
cat_estufa_chi = {'1.0' : 1, '1' : 1, 1 : 1,
                   '2.0': 2, '2' : 2, 2 : 2,
                   ' ': '&'}
cat_estufa_chi

{' ': '&', '1': 1, 1: 1, '1.0': 1, '2': 2, 2: 2, '2.0': 2}

In [92]:
df_copy_vivienda.estufa_chi.replace(cat_estufa_chi, inplace=True)
df_copy_vivienda.estufa_chi.value_counts(sort=True, ascending=False)

&    98259
2    20086
1     5572
Name: estufa_chi, dtype: int64

#### Eliminación de basura `eli_basura`
Definición: Formas de eliminar la basura de la vivienda.
    
Valor Etiqueta
+ 1 La recoge un camión o carrito de basura
+ 2 La tiran en el basurero público
+ 3 La tiran en un contenedor o depósito
+ 4 La queman
+ 5 La entierran
+ 6 La tiran en un terreno baldío o calle
+ 7 La tiran en barranca o grieta
+ 8 La tiran al río, lago o mar

In [93]:
df_copy_vivienda.eli_basura = df_copy_vivienda.eli_basura.astype('category')
df_copy_vivienda.eli_basura.value_counts(sort=True, ascending=False)

1    127954
3     12706
4     12589
2      5121
5      1058
6       807
7       208
8        24
Name: eli_basura, dtype: int64

#### Dispone de lavadero `lavadero`
Definición: Lavadero
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [94]:
df_copy_vivienda.lavadero = df_copy_vivienda.lavadero.astype('category')
df_copy_vivienda.lavadero.value_counts(sort=True, ascending=False)

1    133171
2     27296
Name: lavadero, dtype: int64

#### Dispone de fregadero `fregadero`
Definición: Fregadero o tarja
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [95]:
df_copy_vivienda.fregadero = df_copy_vivienda.fregadero.astype('category')
df_copy_vivienda.fregadero.value_counts(sort=True, ascending=False)

1    96441
2    64026
Name: fregadero, dtype: int64

#### Dispone de regadera `regadera`
Definición: Regadera
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [97]:
df_copy_vivienda.regadera = df_copy_vivienda.regadera.astype('category')
df_copy_vivienda.regadera.value_counts(sort=True, ascending=False)

1    115227
2     45240
Name: regadera, dtype: int64

#### Dispone de regadera eléctrica `rega_elect`
Definición: Regadera eléctrica.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [208]:
df_copy_vivienda.rega_elect = df_copy_vivienda.rega_elect.astype('category')
df_copy_vivienda.rega_elect.value_counts(sort=True, ascending=False)

&      102401
2.0     56155
1.0      1911
Name: rega_elect, dtype: int64

In [209]:
cat_rega_elect = {'1.0' : 1, '1' : 1, 1 : 1,
                   '2.0': 2, '2' : 2, 2 : 2,
                   ' ': '&'}
cat_rega_elect

{' ': '&', '1': 1, 1: 1, '1.0': 1, '2': 2, 2: 2, '2.0': 2}

In [210]:
df_copy_vivienda.rega_elect.replace(cat_rega_elect, inplace=True)
df_copy_vivienda.rega_elect.value_counts(sort=True, ascending=False)

&    102401
2     56155
1      1911
Name: rega_elect, dtype: int64

#### Dispone de tinaco  `tinaco_azo`
Definición: Tinaco en la azotea.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [207]:
df_copy_vivienda.tinaco_azo = df_copy_vivienda.tinaco_azo.astype('category')
df_copy_vivienda.tinaco_azo.value_counts()

1    90384
2    70083
Name: tinaco_azo, dtype: int64

#### Dispone de cisterna  `cisterna`
Definición: Cisterna o aljibe.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [98]:
df_copy_vivienda.cisterna = df_copy_vivienda.cisterna.astype('category')
df_copy_vivienda.cisterna.value_counts(sort=True, ascending=False)

2    129183
1     31284
Name: cisterna, dtype: int64

#### pileta: Dispone de pileta o tanque  `pileta`
Definición: Pileta, tanque o depósito de agua.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [125]:
df_copy_vivienda.pileta = df_copy_vivienda.pileta.astype('category')
df_copy_vivienda.pileta.value_counts(sort=True, ascending=False)

2    91483
1    68984
Name: pileta, dtype: int64

#### Dispone de calentador solar de agua  `calent_sol`
Definición: Calentador de agua (boiler).
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [215]:
df_copy_vivienda.calent_sol = df_copy_vivienda.calent_sol.astype('category')
df_copy_vivienda.calent_sol.value_counts(sort=True, ascending=False)

&      102401
2.0     55368
1.0      2698
Name: calent_sol, dtype: int64

In [217]:
cat_calent_sol = {'1.0' : 1, '1' : 1, 1 : 1,
                   '2.0': 2, '2' : 2, 2 : 2,
                   ' ': '&'}
cat_calent_sol

{' ': '&', '1': 1, 1: 1, '1.0': 1, '2': 2, 2: 2, '2.0': 2}

In [218]:
df_copy_vivienda.calent_sol.replace(cat_calent_sol, inplace=True)
df_copy_vivienda.calent_sol.value_counts(sort=True, ascending=False)

&    102401
2     55368
1      2698
Name: calent_sol, dtype: int64

#### Dispone de calentador de gas para el agua  `calent_gas`
Definición: Calentador de gas u otros.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [216]:
df_copy_vivienda.calent_gas = df_copy_vivienda.calent_gas.astype('category')
df_copy_vivienda.calent_gas.value_counts(sort=True, ascending=False)

&      102401
2.0     34265
1.0     23801
Name: calent_gas, dtype: int64

In [219]:
cat_calent_gas = {'1.0' : 1, '1' : 1, 1 : 1,
                   '2.0': 2, '2' : 2, 2 : 2,
                   ' ': '&'}
cat_calent_gas

{' ': '&', '1': 1, 1: 1, '1.0': 1, '2': 2, 2: 2, '2.0': 2}

In [221]:
df_copy_vivienda.calent_gas.replace(cat_calent_gas, inplace=True)
df_copy_vivienda.calent_gas.value_counts(sort=True, ascending=False)

&    102401
2     34265
1     23801
Name: calent_gas, dtype: int64

#### Dispone de calentador de agua `calentador`
Definición: Calentador de agua (boiler).
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [211]:
#df_copy_vivienda.calentador = df_copy_vivienda.calentador.astype('category')
df_copy_vivienda.calentador.value_counts(sort=True, ascending=False)

&    58066
2    56719
1    45682
Name: calentador, dtype: int64

In [212]:
df_copy_vivienda.calentador = df_copy_vivienda.calentador.astype('category')
df_copy_vivienda.calentador.value_counts(sort=True, ascending=False)

&    58066
2    56719
1    45682
Name: calentador, dtype: int64

In [213]:
cat_calentador = {1.0 : 1, '1.0' : 1, 1 : 1,
                  2.0: 2, '2.0' : 2, 2 : 2}
cat_calentador

{1.0: 1, '1.0': 1, 2.0: 2, '2.0': 2}

In [214]:
df_copy_vivienda.calentador.replace(cat_calentador, inplace=True)
df_copy_vivienda.calentador.value_counts(sort=True, ascending=False)

&    58066
2    56719
1    45682
Name: calentador, dtype: int64

#### Unión de variables `calentador` (2014-2015) -> `calent_gas` (2016)
+ 1 (56719 + 45682) 102401
+ 2 (34265 + 23801) 58066
+ Total 160467

In [268]:
temp_calentador_gas = []
for index in range(len(df_copy_vivienda)):
    if (df_copy_vivienda.calentador[index] == 1) & (df_copy_vivienda.calent_gas[index] == 1):
        temp_calentador_gas.append('1')
    elif (df_copy_vivienda.calentador[index] == 1) & (df_copy_vivienda.calent_gas[index] == '&'):
        temp_calentador_gas.append('1')
    elif (df_copy_vivienda.calentador[index] == '&') & (df_copy_vivienda.calent_gas[index] == 1):
        temp_calentador_gas.append('1')
    elif (df_copy_vivienda.calentador[index] == 2) & (df_copy_vivienda.calent_gas[index] == 2):
        temp_calentador_gas.append('2')
    elif (df_copy_vivienda.calentador[index] == 2) & (df_copy_vivienda.calent_gas[index] == '&'):
        temp_calentador_gas.append('2')
    elif (df_copy_vivienda.calentador[index] == '&') & (df_copy_vivienda.calent_gas[index] == 2):
        temp_calentador_gas.append('2')
    elif (df_copy_vivienda.calentador[index] == '&') & (df_copy_vivienda.calent_gas[index] == '&'):
        temp_calentador_gas.append('&')

In [272]:
df_copy_vivienda.calent_gas = temp_calentador_gas
df_copy_vivienda.calent_gas.value_counts()

2    90984
1    69483
Name: calent_gas, dtype: int64

#### Dispone de medidor de luz `medidor_luz`
Definición: Medidor de luz.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [167]:
df_copy_vivienda.medidor_luz = df_copy_vivienda.medidor_luz.astype('category')
df_copy_vivienda.medidor_luz.value_counts(sort=True, ascending=False)

1    149688
2     10779
Name: medidor_luz, dtype: int64

#### Dispone de bomba de agua `bomba_agua`
Definición: Bomba de agua.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [168]:
df_copy_vivienda.bomba_agua = df_copy_vivienda.bomba_agua.astype('category')
df_copy_vivienda.bomba_agua.value_counts(sort=True, ascending=False)

2    122926
1     37541
Name: bomba_agua, dtype: int64

#### Dispone de tanque de gas estacionario `tanque_gas`
Definición: Tanque de gas estacionario.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [169]:
df_copy_vivienda.tanque_gas = df_copy_vivienda.tanque_gas.astype('category')
df_copy_vivienda.tanque_gas.value_counts(sort=True, ascending=False)

2    144331
1     16136
Name: tanque_gas, dtype: int64

#### Dispone de aire acondicionado `aire_acond`
Definición: Aire acondicionado.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [170]:
df_copy_vivienda.aire_acond = df_copy_vivienda.aire_acond.astype('category')
df_copy_vivienda.aire_acond.value_counts(sort=True, ascending=False)

2    130717
1     29750
Name: aire_acond, dtype: int64

#### Dispone de calefacción `calefacc`
Definición: Calefacción.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [171]:
df_copy_vivienda.calefacc = df_copy_vivienda.calefacc.astype('category')
df_copy_vivienda.calefacc.value_counts(sort=True, ascending=False)

2    157091
1      3376
Name: calefacc, dtype: int64

#### Dispone de chimenea `chimenea`
Definición: Chimenea o calentón de leña.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [172]:
df_copy_vivienda.chimenea = df_copy_vivienda.chimenea.astype('category')
df_copy_vivienda.chimenea.value_counts(sort=True, ascending=False)

2    155492
1      4975
Name: chimenea, dtype: int64

#### Reparar las paredes `repar_pard`
Definición: Necesidad de reparar las paredes.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [173]:
df_copy_vivienda.repar_pard = df_copy_vivienda.repar_pard.astype('category')
df_copy_vivienda.repar_pard.value_counts(sort=True, ascending=False)

2    84409
1    76058
Name: repar_pard, dtype: int64

#### Reparar el techo `repar_tech`
Definición: Necesidad de reparar el techo.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [273]:
df_copy_vivienda.repar_tech = df_copy_vivienda.repar_tech.astype('category')
df_copy_vivienda.repar_tech.value_counts()

2    83518
1    76949
Name: repar_tech, dtype: int64

#### Reparar las tuberías del agua `repar_agua`
Definición: Necesidad de reparar las tuberías o llaves de agua.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [274]:
df_copy_vivienda.repar_agua = df_copy_vivienda.repar_agua.astype('category')
df_copy_vivienda.repar_agua.value_counts()

2    133586
1     26881
Name: repar_agua, dtype: int64

#### Reparar las tuberías del drenaje `repar_dren`
Definición: Necesidad de reparar las tuberías del drenaje o instalaciones de excusado.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [275]:
df_copy_vivienda.repar_dren = df_copy_vivienda.repar_dren.astype('category')
df_copy_vivienda.repar_dren.value_counts()

2    140669
1     19798
Name: repar_dren, dtype: int64

#### Reparar el cableado eléctrico `repar_cabl`
Definición: Necesidad de reparar el cableado eléctrico.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [276]:
df_copy_vivienda.repar_cabl = df_copy_vivienda.repar_cabl.astype('category')
df_copy_vivienda.repar_cabl.value_counts()

2    125784
1     34683
Name: repar_cabl, dtype: int64

#### Construir dormitorio `const_dorm`
Definición: Necesidad de construir un cuarto para dormir.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [277]:
df_copy_vivienda.const_dorm = df_copy_vivienda.const_dorm.astype('category')
df_copy_vivienda.const_dorm.value_counts()

2    91171
1    69296
Name: const_dorm, dtype: int64

#### Construir cocina `const_coci`
Definición: Necesidad de construir un cuarto para cocinar.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [278]:
df_copy_vivienda.const_coci = df_copy_vivienda.const_coci.astype('category')
df_copy_vivienda.const_coci.value_counts()

2    113302
1     47165
Name: const_coci, dtype: int64

In [277]:
df_copy_vivienda.const_dorm = df_copy_vivienda.const_dorm.astype('category')
df_copy_vivienda.const_dorm.value_counts()

2    91171
1    69296
Name: const_dorm, dtype: int64

#### Construir baño `const_bano`
Definición: Necesidad de construir un cuarto de baño.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [282]:
df_copy_vivienda.const_bano = df_copy_vivienda.const_bano.astype('category')
df_copy_vivienda.const_bano.value_counts()

2    103176
1     57291
Name: const_bano, dtype: int64

#### Construir negocio `const_neg`
Definición: Necesidad de construir un cuarto para un negocio.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [279]:
df_copy_vivienda.const_neg = df_copy_vivienda.const_neg.astype('category')
df_copy_vivienda.const_neg.value_counts()

2.0    88072
&      43786
1.0    28609
Name: const_neg, dtype: int64

In [280]:
cat_const_neg = {1.0 : 1, '1.0' : 1, 1 : 1,
                  2.0: 2, '2.0' : 2, 2 : 2}
cat_const_neg

{1.0: 1, '1.0': 1, 2.0: 2, '2.0': 2}

In [281]:
df_copy_vivienda.const_neg.replace(cat_const_neg, inplace=True)
df_copy_vivienda.const_neg.value_counts(sort=True, ascending=False)

2    88072
&    43786
1    28609
Name: const_neg, dtype: int64

#### Tipo de tenencia de la vivienda `tenencia`
Definición: Dominio o posesión legal de la vivienda por sus ocupantes.
    
Valor Etiqueta
+ 1 Es rentada
+ 2 Es prestada
+ 3 Es propia pero la están pagando
+ 4 Es propia
+ 5 Está intestada o en litigio
+ 6 Otra situación

In [283]:
df_copy_vivienda.tenencia = df_copy_vivienda.tenencia.astype('category')
df_copy_vivienda.tenencia.value_counts()

4    89460
2    24382
1    22113
3    18466
5     3660
6     2386
Name: tenencia, dtype: int64

#### Pago de renta de vivienda `pago_renta`
Definición: Pago de renta al mes por la vivienda habitada.
    
Valor Etiqueta
+ 1+

In [285]:
df_copy_vivienda.pago_renta = df_copy_vivienda.pago_renta.astype('category')
df_copy_vivienda.pago_renta.value_counts().head()

&       152341
1000       912
1500       873
2000       650
1200       566
Name: pago_renta, dtype: int64

#### Refrigerador en vivienda rentada `ren_refri` : 
Definición: La vivienda en renta ya contaba con refrigerador.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [287]:
df_copy_vivienda.ren_refri = df_copy_vivienda.ren_refri.astype('category')
df_copy_vivienda.ren_refri.value_counts()

&    152341
2      7739
1       387
Name: ren_refri, dtype: int64

#### Lavadora en vivienda rentada `ren_lavad`
Definición: La vivienda en renta ya contaba con lavadora.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [288]:
df_copy_vivienda.ren_lavad = df_copy_vivienda.ren_lavad.astype('category')
df_copy_vivienda.ren_lavad.value_counts()

&    152341
2      7954
1       172
Name: ren_lavad, dtype: int64

#### Televisión en vivienda rentada `ren_tv`
Definición: La vivienda en renta ya contaba con televisión.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [289]:
df_copy_vivienda.ren_tv = df_copy_vivienda.ren_tv.astype('category')
df_copy_vivienda.ren_tv.value_counts()

&    152341
2      7928
1       198
Name: ren_tv, dtype: int64

#### Horno de microondas en vivienda rentada `ren_horno`
Definición: La vivienda en renta ya contaba con horno de microondas. 
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [290]:
df_copy_vivienda.ren_horno = df_copy_vivienda.ren_horno.astype('category')
df_copy_vivienda.ren_horno.value_counts()

&    152341
2      7976
1       150
Name: ren_horno, dtype: int64

#### Plancha en vivienda rentada `ren_planch`
Definición: La vivienda en renta ya contaba con plancha.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [291]:
df_copy_vivienda.ren_planch = df_copy_vivienda.ren_planch.astype('category')
df_copy_vivienda.ren_planch.value_counts()

&    152341
2      8024
1       102
Name: ren_planch, dtype: int64

#### Estufa en vivienda rentada `ren_estufa`
Definición: La vivienda en renta ya contaba con estufa.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [292]:
df_copy_vivienda.ren_estufa = df_copy_vivienda.ren_estufa.astype('category')
df_copy_vivienda.ren_estufa.value_counts()

&    152341
2      7481
1       645
Name: ren_estufa, dtype: int64

#### Tipo de adquisición `tipo_adqui`
Definición: Situación de la vivienda al momento de adquirirla.
    
Valor Etiqueta
+ 1 La compró hecha
+ 2 Compró el terreno y construyó
+ 3 Ocupó el terreno y construyó
+ 4 Heredó el terreno y construyó
+ 5 Heredó la vivienda hecha
+ 6 Le regalaron el terreno
+ 7 Le regalaron la vivienda hecha
+ 8 Ocupó la vivienda

In [293]:
df_copy_vivienda.tipo_adqui = df_copy_vivienda.tipo_adqui.astype('category')
df_copy_vivienda.tipo_adqui.value_counts()

&      52541
2      27500
1      25605
2.0    12576
4      10256
1.0     9904
6       6440
3.0     5779
5       5115
4.0     2632
7       1229
3        857
8         33
Name: tipo_adqui, dtype: int64

In [295]:
cat_tipo_adqui = {'1.0' : 1, '1' : 1, 1.0 : 1,
                  '2.0': 2, '2' : 2, 2.0 : 2,
                  '3.0': 3, '2' : 2, 3.0 : 2,
                  '4.0': 4, '2' : 2, 4.0 : 2,
                  '5.0': 5, '2' : 2, 5.0 : 2,
                  '2.0': 6, '2' : 2, 6.0 : 2,
                  '2.0': 7, '2' : 2, 7.0 : 2,
                  '2.0': 8, '2' : 2, 8.0 : 2}
cat_tipo_adqui

{'1': 1,
 '1.0': 1,
 1.0: 1,
 '2': 2,
 '2.0': 8,
 2.0: 2,
 '3.0': 3,
 3.0: 2,
 '4.0': 4,
 4.0: 2,
 '5.0': 5,
 5.0: 2,
 6.0: 2,
 7.0: 2,
 8.0: 2}

In [296]:
df_copy_vivienda.tipo_adqui.replace(cat_tipo_adqui, inplace=True)
df_copy_vivienda.tipo_adqui.value_counts(sort=True, ascending=False)

&    52541
2    48487
1    35509
4    10256
6     6440
5     5115
7     1229
3      857
8       33
Name: tipo_adqui, dtype: int64

#### Compró la vivienda usada `viv_usada`
Definición: Estado de la vivienda al momento de adquirirla.
    
Valor Etiqueta
+ 1 Sí
+ 2 No

In [297]:
df_copy_vivienda.viv_usada = df_copy_vivienda.viv_usada.astype('category')
df_copy_vivienda.viv_usada.value_counts()

&      118614
1       16176
2       15679
2.0      5978
1.0      3926
9          94
Name: viv_usada, dtype: int64

In [298]:
cat_viv_usada = {'1.0' : 1, '1' : 1, 1.0 : 1,
                  '2.0': 2, '2' : 2, 2.0 : 2,
                  '3.0': 3, '2' : 2, 3.0 : 2,
                  '4.0': 4, '2' : 2, 4.0 : 2,
                  '5.0': 5, '2' : 2, 5.0 : 2,
                  '2.0': 6, '2' : 2, 6.0 : 2,
                  '2.0': 7, '2' : 2, 7.0 : 2,
                  '2.0': 8, '2' : 2, 8.0 : 2,
                  '9.0': '&', '9' : '&', 9.0 : '&'}
cat_viv_usada

{'1': 1,
 '1.0': 1,
 1.0: 1,
 '2': 2,
 '2.0': 8,
 2.0: 2,
 '3.0': 3,
 3.0: 2,
 '4.0': 4,
 4.0: 2,
 '5.0': 5,
 5.0: 2,
 6.0: 2,
 7.0: 2,
 8.0: 2,
 '9': '&',
 '9.0': '&',
 9.0: '&'}

In [299]:
df_copy_vivienda.viv_usada.replace(cat_viv_usada, inplace=True)
df_copy_vivienda.viv_usada.value_counts(sort=True, ascending=False)

&    118708
2     21657
1     20102
Name: viv_usada, dtype: int64

#### Recursos propios `financia_1`
Definición: Para pagar o construir esta vivienda el dueño uso sus propios recursos.
    
Valor Etiqueta
+ 1 Sí
+ & representa un dato “No especificado”.

In [300]:
df_copy_vivienda.financia_1 = df_copy_vivienda.financia_1.astype('category')
df_copy_vivienda.financia_1.value_counts()

&    111109
1     49358
Name: financia_1, dtype: int64

#### Apoyo de FONHAPO `financia_2`
Definición: Para pagar o construir esta vivienda el dueño tuvo un apoyo de FONHAPO.
    
Valor Etiqueta
+ 2 Sí
+ & representa un dato “No especificado”.

In [301]:
df_copy_vivienda.financia_2 = df_copy_vivienda.financia_2.astype('category')
df_copy_vivienda.financia_2.value_counts()

&    159909
2       558
Name: financia_2, dtype: int64

#### Crédito INFONAVIT o FOVISSSTE `financia_3`
Definición: Para pagar o construir esta vivienda el dueño tuvo un crédito de INFONAVIT o FOVISSSTE.
    
Valor Etiqueta
+ 3 Sí
+ & representa un dato “No especificado”.

In [308]:
df_copy_vivienda.financia_3 = df_copy_vivienda.financia_3.astype('category')
df_copy_vivienda.financia_3.value_counts()

&    145409
3     15058
Name: financia_3, dtype: int64

#### Crédito bancario `financia_4`
Definición: Para pagar o construir esta vivienda el dueño tuvo un crédito bancario.
    
Valor Etiqueta
+ 4 Sí
+ & representa un dato “No especificado”.

In [303]:
df_copy_vivienda.financia_4 = df_copy_vivienda.financia_4.astype('category')
df_copy_vivienda.financia_4.value_counts()

&    156803
4      3664
Name: financia_4, dtype: int64

#### Crédito microfinanciero `financia_5`
Definición: Para pagar o construir esta vivienda el dueño tuvo un crédito de una microfinanciera.
    
Valor Etiqueta
+ 5 Sí
+ & representa un dato “No especificado”.

In [304]:
df_copy_vivienda.financia_5 = df_copy_vivienda.financia_5.astype('category')
df_copy_vivienda.financia_5.value_counts()

&    160072
5       395
Name: financia_5, dtype: int64

#### Crédito caja de ahorro `financia_6`
Definición: Para pagar o construir esta vivienda el dueño tuvo un crédito de una caja de ahorro.
    
Valor Etiqueta
+ 6 Sí
+ & representa un dato “No especificado”.

In [305]:
df_copy_vivienda.financia_6 = df_copy_vivienda.financia_6.astype('category')
df_copy_vivienda.financia_6.value_counts()

&    159698
6       769
Name: financia_6, dtype: int64

#### Crédito de otra institución `financia_7`
Definición: Para pagar o construir esta vivienda el dueño tuvo crédito de otra institución.
    
Valor Etiqueta
+ 7 Sí
+ & representa un dato “No especificado”.

In [306]:
df_copy_vivienda.financia_7 = df_copy_vivienda.financia_7.astype('category')
df_copy_vivienda.financia_7.value_counts()

&    158240
7      2227
Name: financia_7, dtype: int64

#### Préstamo familiar `financia_8`
Definición: Para pagar o construir esta vivienda el dueño tuvo préstamo de un familiar, amigo o prestamista.
    
Valor Etiqueta
+ 8 Sí
+ & representa un dato “No especificado”.

In [307]:
df_copy_vivienda.financia_8 = df_copy_vivienda.financia_8.astype('category')
df_copy_vivienda.financia_8.value_counts()

&    158050
8      2417
Name: financia_8, dtype: int64

+ 1     49358
+ 2       558
+ 3     15058
+ 4      3664
+ 5       395
+ 6       769
+ 7      2227
+ Total 72029

#### Valor de la vivienda `valor_viv`
Definición: A cuánto asciende el valor de la vivienda habitada.
    
Valor Etiqueta
+ 1 Menos de $50,000

* 2 De' $50,001' a $100,000.00

* 3 De $100,001.00 a $300,000.00

+ 4 De $300,001.00 a $500,000.00

+ 5 De $500,001.00 a $1,000,000.00

+ 6 De $1,000,001.00 a $2,000,000.00

+ 7 Más de $2,000,001.00
+ & Representa un dato “No especificado”

In [309]:
df_copy_vivienda.valor_viv = df_copy_vivienda.valor_viv.astype('category')
df_copy_vivienda.valor_viv.value_counts()

&    83432
9    28439
3    14400
4    12137
5     7955
2     6649
1     3465
6     2842
7     1148
Name: valor_viv, dtype: int64

#### Identificador del primer dueño `num_dueno1`
Definición: Número de renglón del dueño o dueña de la vivienda.
    
Valor Etiqueta
+ 1+
+ & Representa un dato “No especificado”

In [320]:
df_copy_vivienda.num_dueno1 = df_copy_vivienda.num_dueno1.astype('category')
df_copy_vivienda.num_dueno1.value_counts()

01      62432
&       52966
1.0     25433
02      12571
2.0      4881
03       1102
3.0       391
04        259
05        124
4.0        85
06         73
5.0        55
6.0        29
07         27
08         13
7.0         7
09          4
8.0         3
9.0         3
10.0        3
10          2
13          2
11.0        1
15          1
Name: num_dueno1, dtype: int64

In [325]:
cat_num_dueno1 = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1, '01' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2, '02' : 3,
                 3.0: 3, '3.0' : 3, '3' : 3, '03' : 3,
                 4.0: 4, '4.0' : 4, '4': 4, '04' : 4,
                 5.0: 4, '5.0' : 5, '5': 5, '05' : 5,
                 6.0: 4, '6.0' : 6, '6': 6, '06' : 6,
                 7.0: 4, '7.0' : 7, '7': 7, '07' : 7,
                 8.0: 4, '8.0' : 8, '8': 8, '08' : 8,
                 9.0: 4, '9.0' : 9, '9': 9, '09' : 9,
                 10.0: 4, '10.0' : 10, '10': 10, '10' : 10,
                 11.0: 4, '11.0' : 11, '11': 11, '11' : 11,
                 12.0: 4, '12.0' : 12, '12': 12, '12' : 12,
                 13.0: 4, '13.0' : 13, '13': 13, '13' : 13,
                 14.0: 4, '14.0' : 14, '14': 14, '14' : 14,
                 15.0: 4, '15.0' : 15, '15': 15, '15' : 15,}
cat_num_dueno1

{'01': 1,
 '02': 3,
 '03': 3,
 '04': 4,
 '05': 5,
 '06': 6,
 '07': 7,
 '08': 8,
 '09': 9,
 '1': 1,
 1.0: 1,
 '1.0': 1,
 '10': 10,
 10.0: 4,
 '10.0': 10,
 '11': 11,
 11.0: 4,
 '11.0': 11,
 '12': 12,
 12.0: 4,
 '12.0': 12,
 '13': 13,
 13.0: 4,
 '13.0': 13,
 '14': 14,
 14.0: 4,
 '14.0': 14,
 '15': 15,
 15.0: 4,
 '15.0': 15,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '3': 3,
 3.0: 3,
 '3.0': 3,
 '4': 4,
 4.0: 4,
 '4.0': 4,
 '5': 5,
 5.0: 4,
 '5.0': 5,
 '6': 6,
 6.0: 4,
 '6.0': 6,
 '7': 7,
 7.0: 4,
 '7.0': 7,
 '8': 8,
 8.0: 4,
 '8.0': 8,
 '9': 9,
 9.0: 4,
 '9.0': 9}

In [326]:
df_copy_vivienda.num_dueno1.replace(cat_num_dueno1, inplace=True)
df_copy_vivienda.num_dueno1.value_counts(sort=True, ascending=False)

1     87865
&     52966
3     14064
2      4881
4       445
5       124
6        73
7        27
8        13
9         4
13        2
10        2
15        1
Name: num_dueno1, dtype: int64

#### Hogar del primer dueño `hog_dueno1`
Definición: Número de hogar del dueño o dueña de la vivienda.
    
Valor Etiqueta
+ 1+
+ & Representa un dato “No especificado”

In [310]:
df_copy_vivienda.hog_dueno1 = df_copy_vivienda.hog_dueno1.astype('category')
df_copy_vivienda.hog_dueno1.value_counts()

1      76393
&      52967
1.0    30809
2        201
2.0       74
3         12
3.0        8
4          3
Name: hog_dueno1, dtype: int64

In [316]:
cat_hog_dueno1 = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2,
                 3.0: 3, '3.0' : 3, '3' : 3,
                 4.0: 4, '4.0' : 4, '4': 4}
cat_hog_dueno1

{'1': 1,
 1.0: 1,
 '1.0': 1,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '3': 3,
 3.0: 3,
 '3.0': 3,
 '4': 4,
 4.0: 4,
 '4.0': 4}

In [317]:
df_copy_vivienda.hog_dueno1.replace(cat_hog_dueno1, inplace=True)
df_copy_vivienda.hog_dueno1.value_counts(sort=True, ascending=False)

1    107202
&     52967
2       275
3        20
4         3
Name: hog_dueno1, dtype: int64

#### Identificador del segundo dueño `num_dueno2`
Definición: Número de renglón del dueño o dueña de la vivienda.
    
Valor Etiqueta
+ 1+
+ & Representa un dato “No especificado”

In [324]:
df_copy_vivienda.num_dueno2 = df_copy_vivienda.num_dueno2.astype('category')
df_copy_vivienda.num_dueno2.value_counts()

&      150503
02       6521
2.0      1937
01        724
1.0       414
03        176
04         70
3.0        48
4.0        25
05         23
06         10
5.0         6
07          3
6.0         2
08          2
09          1
9.0         1
12          1
Name: num_dueno2, dtype: int64

In [327]:
cat_num_dueno2 = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1, '01' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2, '02' : 3,
                 3.0: 3, '3.0' : 3, '3' : 3, '03' : 3,
                 4.0: 4, '4.0' : 4, '4': 4, '04' : 4,
                 5.0: 4, '5.0' : 5, '5': 5, '05' : 5,
                 6.0: 4, '6.0' : 6, '6': 6, '06' : 6,
                 7.0: 4, '7.0' : 7, '7': 7, '07' : 7,
                 8.0: 4, '8.0' : 8, '8': 8, '08' : 8,
                 9.0: 4, '9.0' : 9, '9': 9, '09' : 9,
                 10.0: 4, '10.0' : 10, '10': 10, '10' : 10,
                 11.0: 4, '11.0' : 11, '11': 11, '11' : 11,
                 12.0: 4, '12.0' : 12, '12': 12, '12' : 12,
                 13.0: 4, '13.0' : 13, '13': 13, '13' : 13,
                 14.0: 4, '14.0' : 14, '14': 14, '14' : 14,
                 15.0: 4, '15.0' : 15, '15': 15, '15' : 15,}
cat_num_dueno2

{'01': 1,
 '02': 3,
 '03': 3,
 '04': 4,
 '05': 5,
 '06': 6,
 '07': 7,
 '08': 8,
 '09': 9,
 '1': 1,
 1.0: 1,
 '1.0': 1,
 '10': 10,
 10.0: 4,
 '10.0': 10,
 '11': 11,
 11.0: 4,
 '11.0': 11,
 '12': 12,
 12.0: 4,
 '12.0': 12,
 '13': 13,
 13.0: 4,
 '13.0': 13,
 '14': 14,
 14.0: 4,
 '14.0': 14,
 '15': 15,
 15.0: 4,
 '15.0': 15,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '3': 3,
 3.0: 3,
 '3.0': 3,
 '4': 4,
 4.0: 4,
 '4.0': 4,
 '5': 5,
 5.0: 4,
 '5.0': 5,
 '6': 6,
 6.0: 4,
 '6.0': 6,
 '7': 7,
 7.0: 4,
 '7.0': 7,
 '8': 8,
 8.0: 4,
 '8.0': 8,
 '9': 9,
 9.0: 4,
 '9.0': 9}

In [328]:
df_copy_vivienda.num_dueno2.replace(cat_num_dueno2, inplace=True)
df_copy_vivienda.num_dueno2.value_counts(sort=True, ascending=False)

&     150503
3       6745
2       1937
1       1138
4        104
5         23
6         10
7          3
8          2
12         1
9          1
Name: num_dueno2, dtype: int64

#### Hogar del segundo dueño `hog_dueno2` 
Definición: Número de hogar del dueño o dueña de la vivienda.
    
Valor Etiqueta
+ 1+
+ & Representa un dato “No especificado”

In [313]:
df_copy_vivienda.hog_dueno2 = df_copy_vivienda.hog_dueno2.astype('category')
df_copy_vivienda.hog_dueno2.value_counts()

&      150503
1        7513
1.0      2423
2          17
2.0         9
3           1
3.0         1
Name: hog_dueno2, dtype: int64

In [329]:
cat_hog_dueno2 = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2,
                 3.0: 3, '3.0' : 3, '3' : 3,
                 4.0: 4, '4.0' : 4, '4': 4}
cat_hog_dueno2

{'1': 1,
 1.0: 1,
 '1.0': 1,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '3': 3,
 3.0: 3,
 '3.0': 3,
 '4': 4,
 4.0: 4,
 '4.0': 4}

In [330]:
df_copy_vivienda.hog_dueno2.replace(cat_hog_dueno2, inplace=True)
df_copy_vivienda.hog_dueno2.value_counts(sort=True, ascending=False)

&    150503
1      9936
2        26
3         2
Name: hog_dueno2, dtype: int64

#### Escrituras de la vivienda `escrituras`
Definición: Escrituras o título de propiedad de la vivienda.
    
Valor Etiqueta
+ 1 Escritura
+ 2 Certificado de posesión
+ 3 Título de propiedad
+ 4 Contrato de compra venta
+ 5 No se cuenta con documento
+ & Representa un dato “No especificado”

In [331]:
df_copy_vivienda.escrituras = df_copy_vivienda.escrituras.astype('category')
df_copy_vivienda.escrituras.value_counts()

&      52541
1      47507
1.0    22045
5       8551
3       7592
2       7541
3.0     6042
4       4738
2.0     2529
9       1106
9.0      275
Name: escrituras, dtype: int64

In [333]:
cat_escrituras = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2,
                 3.0: 3, '3.0' : 3, '3' : 3,
                 4.0: 4, '4.0' : 4, '4': 4,
                 5.0: 5, '5.0' : 5, '5': 5,
                 9.0: '&', '9.0' : '&', '9': '&'}
cat_escrituras

{'1': 1,
 1.0: 1,
 '1.0': 1,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '3': 3,
 3.0: 3,
 '3.0': 3,
 '4': 4,
 4.0: 4,
 '4.0': 4,
 '5': 5,
 5.0: 5,
 '5.0': 5,
 '9': '&',
 9.0: '&',
 '9.0': '&'}

In [334]:
df_copy_vivienda.escrituras.replace(cat_escrituras, inplace=True)
df_copy_vivienda.escrituras.value_counts(sort=True, ascending=False)

1    69552
&    53922
3    13634
2    10070
5     8551
4     4738
Name: escrituras, dtype: int64

#### Venta o herencia de la vivienda `vend_hered`
Definición: Venta o herencia de la vivienda.
    
Valor Etiqueta
+ 1 1 Sí
+ 2 No
+ & Representa un dato “No especificado”

In [335]:
df_copy_vivienda.vend_hered = df_copy_vivienda.vend_hered.astype('category')
df_copy_vivienda.vend_hered.value_counts()

&    83432
1    76704
2      331
Name: vend_hered, dtype: int64

#### Identificador persona 1 con facultad de venta `facul_no1`
Definición: Identificador de la primera persona que tiene facultad de vender o heredar la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [336]:
df_copy_vivienda.facul_no1 = df_copy_vivienda.facul_no1.astype('category')
df_copy_vivienda.facul_no1.value_counts()

&     160273
01       115
02        65
03         9
04         5
Name: facul_no1, dtype: int64

#### Hogar persona 1 con facultad de venta `facul_nh1` : 
Definición: Hogar de la primer persona con facultad de vender o heredar la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [337]:
df_copy_vivienda.facul_nh1 = df_copy_vivienda.facul_nh1.astype('category')
df_copy_vivienda.facul_nh1.value_counts()

&    160273
1       194
Name: facul_nh1, dtype: int64

#### Identificador persona 2 con facultad de venta `facul_no2`
Definición: Identificador de la segunda persona que tiene facultad de vender o heredar la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [338]:
df_copy_vivienda.facul_no2 = df_copy_vivienda.facul_no2.astype('category')
df_copy_vivienda.facul_no2.value_counts()

&     160412
02        37
01        13
04         3
03         2
Name: facul_no2, dtype: int64

#### Hogar persona 2 con facultad de venta `facul_nh2` : 
Definición: Hogar de la segunda persona con facultad de vender o heredar la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [339]:
df_copy_vivienda.facul_nh2 = df_copy_vivienda.facul_nh2.astype('category')
df_copy_vivienda.facul_nh2.value_counts()

&    160411
1        55
2         1
Name: facul_nh2, dtype: int64

#### No sabe facultad de venta `facul_ne` 
Definición: No sabe quién es la persona con facultad de vender o heredar la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [341]:
df_copy_vivienda.facul_ne = df_copy_vivienda.facul_ne.astype('category')
df_copy_vivienda.facul_ne.value_counts()

&    160335
9       132
Name: facul_ne, dtype: int64

#### Identificador persona 1 con poder de disposición de dinero `dispos_no1`
Definición: Identificador de la primer persona que tiene el poder de disposición del dinero por la venta o renta de la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [342]:
df_copy_vivienda.dispos_no1 = df_copy_vivienda.dispos_no1.astype('category')
df_copy_vivienda.dispos_no1.value_counts()

&     160254
01       135
02        62
03        13
04         3
Name: dispos_no1, dtype: int64

#### Hogar persona 1 con poder de disposición de dinero `dispos_nh1` 
Definición: Hogar de la primer persona que tiene el poder de disposición del dinero por la venta o renta de la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [343]:
df_copy_vivienda.dispos_nh1 = df_copy_vivienda.dispos_nh1.astype('category')
df_copy_vivienda.dispos_nh1.value_counts()

&    160254
1       213
Name: dispos_nh1, dtype: int64

#### Identificador persona 2 con poder de disposición de dinero `dispos_no2`
Definición: Identificador de la segunda persona que tiene el poder de disposición del dinero por la venta o renta de la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [344]:
df_copy_vivienda.dispos_no2 = df_copy_vivienda.dispos_no2.astype('category')
df_copy_vivienda.dispos_no2.value_counts()

&     160404
02        43
01        14
04         3
03         3
Name: dispos_no2, dtype: int64

#### Hogar persona 2 con poder de disposición de dinero `dispos_nh2` 
Definición: Hogar de la segunda persona que tiene el poder de disposición del dinero por la venta o renta de la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [345]:
df_copy_vivienda.dispos_nh2 = df_copy_vivienda.dispos_nh2.astype('category')
df_copy_vivienda.dispos_nh2.value_counts()

&    160404
1        62
2         1
Name: dispos_nh2, dtype: int64

#### No sabe poder de disposición `dispos_ne` 
Definición: No sabe quien es la persona que tiene el poder de disposición del dinero por la venta o renta de la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

In [346]:
df_copy_vivienda.dispos_ne = df_copy_vivienda.dispos_ne.astype('category')
df_copy_vivienda.dispos_ne.value_counts()

&    160352
9       115
Name: dispos_ne, dtype: int64

#### Disponibilidad de computadora  `computador`
Definición: Dispone de computadora en la vivienda.
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & Representa un dato “No especificado”

In [347]:
df_copy_vivienda.computador = df_copy_vivienda.computador.astype('category')
df_copy_vivienda.computador.value_counts()

&      102401
2.0     36441
1.0     21625
Name: computador, dtype: int64

In [348]:
cat_computador = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2,                 
                 9.0: '&', '9.0' : '&', '9': '&'}
cat_computador

{'1': 1,
 1.0: 1,
 '1.0': 1,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '9': '&',
 9.0: '&',
 '9.0': '&'}

In [349]:
df_copy_vivienda.computador.replace(cat_computador, inplace=True)
df_copy_vivienda.computador.value_counts(sort=True, ascending=False)

&    102401
2     36441
1     21625
Name: computador, dtype: int64

#### Disponibilidad de teléfono celular `celular`
Definición: Dispone de teléfono celular en la vivienda.
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & Representa un dato “No especificado”

In [350]:
df_copy_vivienda.celular = df_copy_vivienda.celular.astype('category')
df_copy_vivienda.celular.value_counts()

&      102401
2.0     37568
1.0     20498
Name: celular, dtype: int64

In [351]:
cat_celular = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2,                 
                 9.0: '&', '9.0' : '&', '9': '&'}
cat_celular

{'1': 1,
 1.0: 1,
 '1.0': 1,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '9': '&',
 9.0: '&',
 '9.0': '&'}

In [352]:
df_copy_vivienda.celular.replace(cat_celular, inplace=True)
df_copy_vivienda.celular.value_counts(sort=True, ascending=False)

&    102401
2     37568
1     20498
Name: celular, dtype: int64

#### Disponibilidad de línea telefónica fija `tel_fijo`
Definición: Dispone de línea telefónica fija en la vivienda.
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & Representa un dato “No especificado”

In [353]:
df_copy_vivienda.tel_fijo = df_copy_vivienda.tel_fijo.astype('category')
df_copy_vivienda.tel_fijo.value_counts()

&      102401
2.0     32124
1.0     25942
Name: tel_fijo, dtype: int64

In [354]:
cat_tel_fijo = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2,                 
                 9.0: '&', '9.0' : '&', '9': '&'}
cat_tel_fijo

{'1': 1,
 1.0: 1,
 '1.0': 1,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '9': '&',
 9.0: '&',
 '9.0': '&'}

In [355]:
df_copy_vivienda.tel_fijo.replace(cat_tel_fijo, inplace=True)
df_copy_vivienda.tel_fijo.value_counts(sort=True, ascending=False)

&    102401
2     32124
1     25942
Name: tel_fijo, dtype: int64

#### Disponibilidad de internet `internet`
Definición: Dispone de internet en la vivienda.
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & Representa un dato “No especificado”

In [356]:
df_copy_vivienda.internet = df_copy_vivienda.internet.astype('category')
df_copy_vivienda.internet.value_counts()

&      102401
1.0     31229
2.0     26837
Name: internet, dtype: int64

In [357]:
cat_internet = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2,                 
                 9.0: '&', '9.0' : '&', '9': '&'}
cat_internet

{'1': 1,
 1.0: 1,
 '1.0': 1,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '9': '&',
 9.0: '&',
 '9.0': '&'}

In [358]:
df_copy_vivienda.internet.replace(cat_internet, inplace=True)
df_copy_vivienda.internet.value_counts(sort=True, ascending=False)

&    102401
1     31229
2     26837
Name: internet, dtype: int64

#### Disponibilidad de servicio de televisión de paga `tv_paga`
Definición: Dispone de servicio de televisión de paga en la vivienda.
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & Representa un dato “No especificado”

In [359]:
df_copy_vivienda.tv_paga = df_copy_vivienda.tv_paga.astype('category')
df_copy_vivienda.tv_paga.value_counts()

&      102401
1.0     50608
2.0      7458
Name: tv_paga, dtype: int64

In [360]:
cat_tv_paga = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2,                 
                 9.0: '&', '9.0' : '&', '9': '&'}
cat_tv_paga

{'1': 1,
 1.0: 1,
 '1.0': 1,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '9': '&',
 9.0: '&',
 '9.0': '&'}

In [361]:
df_copy_vivienda.tv_paga.replace(cat_tv_paga, inplace=True)
df_copy_vivienda.tv_paga.value_counts(sort=True, ascending=False)

&    102401
1     50608
2      7458
Name: tv_paga, dtype: int64

#### Disponibilidad de automóvil propio `auto_pick`
Definición: Dispone de automóvil o camioneta (propia) en la vivienda.
    
Valor Etiqueta
+ 1 Sí
+ 2 No
+ & Representa un dato “No especificado”

In [362]:
df_copy_vivienda.auto_pick = df_copy_vivienda.auto_pick.astype('category')
df_copy_vivienda.auto_pick.value_counts()

&      102401
2.0     32353
1.0     25713
Name: auto_pick, dtype: int64

In [363]:
cat_auto_pick = {1.0 : 1, '1.0' : 1, 1 : 1, '1' : 1,
                 2.0: 2, '2.0' : 2, 2 : 2, '2' : 2,                 
                 9.0: '&', '9.0' : '&', '9': '&'}
cat_auto_pick

{'1': 1,
 1.0: 1,
 '1.0': 1,
 '2': 2,
 2.0: 2,
 '2.0': 2,
 '9': '&',
 9.0: '&',
 '9.0': '&'}

In [364]:
df_copy_vivienda.auto_pick.replace(cat_auto_pick, inplace=True)
df_copy_vivienda.auto_pick.value_counts(sort=True, ascending=False)

&    102401
2     32353
1     25713
Name: auto_pick, dtype: int64

#### Total de residentes de la vivienda `tot_resid`
Definición: Número de personas que habitan la vivienda.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”

Nota: Variable construida, es la suma de los registros de la tabla PERSONA y del total de trabajadores domésticos.

In [365]:
df_copy_vivienda.tot_resid = df_copy_vivienda.tot_resid.astype('category')
df_copy_vivienda.tot_resid.value_counts()

4     36666
3     31158
2     27013
5     25822
1     15098
6     12327
7      5977
8      2921
9      1587
10      889
11      454
12      228
13      140
14       76
15       53
16       23
17       10
19        8
20        8
18        4
22        2
24        1
21        1
25        1
Name: tot_resid, dtype: int64

#### Área geoestadística básica `ageb`
Definición: Subdivisión de las AGEM y constituye la unidad básica del Marco Geoestadístico Nacional de acuerdo a las características del terreno se clasifican en dos tipos: Área geoestadística básica urbana y Área geoestadística básica rural.
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”


In [368]:
df_copy_vivienda.ageb = df_copy_vivienda.ageb.astype('category')
df_copy_vivienda.ageb.value_counts().head()

005-4    508
002-A    506
002-0    498
001-2    482
005-1    482
Name: ageb, dtype: int64

#### Tamaño de localidad `tam_loc` 
Definición: Tamaño de la localidad por número de habitantes.
    
Valor Etiqueta
+ 1 Localidades con 100 000 y más habitantes
+ 2 Localidades con 15 000 a 99 999 habitantes
+ 3 Localidades con 2 500 a 14 999 habitantes
+ 4 Localidades con menos de 2 500 habitantes
+ & Representa un dato “No especificado”


In [370]:
df_copy_vivienda.tam_loc = df_copy_vivienda.tam_loc.astype('category')
df_copy_vivienda.tam_loc.value_counts()

1    73099
4    37906
3    24893
2    24569
Name: tam_loc, dtype: int64

#### Estrato socioeconómico `est_socio` 
Definición: Clasificación de las viviendas del país de acuerdo a ciertas características socioeconómicas de las personas que las habitan, así como características físicas y el equipamiento de las mismas expresadas por medio de 24 indicadores construidos con información del Censo de Población y Vivienda 2010. Esta estratificación se realizó por medio de métodos estadísticos multivariados.
    
Valor Etiqueta
+ 1 Bajo
+ 2 Medio bajo
+ 3 Medio alto
+ 4 Alto
+ & Representa un dato “No especificado”


In [371]:
df_copy_vivienda.est_socio = df_copy_vivienda.est_socio.astype('category')
df_copy_vivienda.est_socio.value_counts()

2    81079
3    34886
1    29748
4    14754
Name: est_socio, dtype: int64

#### Estrato de diseño muestral `est_dis` 
Definición: Son los estratos seleccionados en la segunda etapa del muestreo y corresponden a grupos de UPMs de acuerdo con su estrato geográfico (entidad-ámbito-zona).
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”


In [373]:
df_copy_vivienda.est_dis = df_copy_vivienda.est_dis.astype('category')
df_copy_vivienda.est_dis.value_counts().head()

2    8669
5    5884
8    5491
9    5079
1    4327
Name: est_dis, dtype: int64

#### Unidad primaria de muestreo `upm`  
Definición: Las unidades primarias de muestreo están constituidas por agrupaciones de viviendas con características diferenciadas dependiendo del ámbito al que pertenecen. Estas unidades son seleccionadas en la primera etapa del muestreo y corresponden a áreas geográficas con límites identificables en el terreno que agrupan un conjunto de viviendas, cuyo número dependedel tamaño de las localidades. 
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”


In [374]:
df_copy_vivienda.upm = df_copy_vivienda.upm.astype('category')
df_copy_vivienda.upm.value_counts().head()

4410    81
4610    75
6400    74
6160    72
4420    71
Name: upm, dtype: int64

#### Factor de expansión `factor`
Definición: El factor de expansión constituye el peso que se le da a cada unidad muestral para generalizar los resultados de la muestra a la población
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”


In [377]:
df_copy_vivienda.factor = df_copy_vivienda.factor.astype('category')
df_copy_vivienda.factor.value_counts().head()

&        102401
134.0       246
118.0       215
196.0       208
178.0       206
Name: factor, dtype: int64

#### Factor de expansión `factor_viv`
Definición: El factor de expansión constituye el peso que se le da a cada unidad muestral para generalizar los resultados de la muestra a la población
    
Valor Etiqueta
+ 1 +
+ & Representa un dato “No especificado”


In [379]:
df_copy_vivienda.factor_viv = df_copy_vivienda.factor_viv.astype('category')
df_copy_vivienda.factor_viv.value_counts().head()

&        58066
237.0      292
166.0      278
216.0      274
227.0      273
Name: factor_viv, dtype: int64

In [380]:
#exportar csv

path = '../../Datasets/ENH/dataframes/vivienda.csv'

df_copy_vivienda.to_csv(path_or_buf=path, index=True, encoding='utf-8')